### ISSUE TITLE
Help: How to apply the unattend.xml

### ISSUE THREAD

**Issue by engineer22** [pbatard/rufus#2291]
Hello all, I know this is not an Issue but I would like your help if it's allowed.

I like playing with ISO images and I'm trying to create my ideal ISO on my own. I have already figured out how to bypass TPM, Secure Boot, RAM, etc., and how to activate Windows 11 using the $OEM$ folder in the Sources parent folder.

What I'm trying to figure out is how to set the **OOBE\BYPASSNRO** in the ISO so I don't have to type it via CMD when installing.

From digging around I found [this](https://github.com/pbatard/rufus/issues/1981) from @pbatard but I don't know how to do this.

I would really appreciate any information on how to add the **OOBE\BYPASSNRO** command in the ISO itself.

TIA, cheers!

**Comment by pineapple63** [comment]
I have not yet checked to make sure that this works properly without all of the additional options i include when creating an unattend file, but i think `<HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>` should be what you need
I would also suggest adding `<HideOnlineAccountScreens>true</HideOnlineAccountScreens>` as that will bypass it even if you accidently leave a network cable connected (if the computer is connected using a wired connection)

it should look something like this:
`<settings pass="oobeSystem">
        <component name="Microsoft-Windows-Shell-Setup" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="nonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
            <OOBE>
                <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
                <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
            </OOBE>
        </component>
    </settings>`

**Comment by pbatard** [comment]
@engineer22, I would encourage you to have a look at https://github.com/AveYo/MediaCreationTool.bat/blob/main/bypass11/AutoUnattend.xml since this is provably what you want.

As such, I will close this issue.

**Comment by engineer22** [comment]
> @engineer22, I would encourage you to have a look at https://github.com/AveYo/MediaCreationTool.bat/blob/main/bypass11/AutoUnattend.xml since this is provably what you want.
> 
> As such, I will close this issue.

Thanks for the info @pbatard do I add this file in the root folder of the ISO? That's what I'm not getting, and I see that I have to put a Product Key.

I already have saved a folder in $OEM$ to activate Windows 11, do I need to change the Product Key in Autounattend.xml to match the one in the OEM folder?

**Comment by pbatard** [comment]
@engineer22, I'm afraid I can not offer support for the use of third party unattend files. You asked for pointers to help you, and I gave you that. If you need more, I would really advise you to post on Windows user forums, or search the internet for the use and application of answer files, as this issue tracker is not a support forum for answer files, sorry.

**Comment by engineer22** [comment]
> I have not yet checked to make sure that this works properly without all of the additional options i include when creating an unattend file, but i think `<HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>` should be what you need
> I would also suggest adding `<HideOnlineAccountScreens>true</HideOnlineAccountScreens>` as that will bypass it even if you accidently leave a network cable connected (if the computer is connected using a wired connection)
> 
> it should look something like this:
> ```xml
>     <settings pass="oobeSystem">
>         <component name="Microsoft-Windows-Shell-Setup" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="nonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
>             <OOBE>
>                 <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
>                 <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
>             </OOBE>
>         </component>
>     </settings>
> ```
> 

Do you add this file in the root of the ISO folder or in the Sources folder?

**Comment by pineapple63** [comment]
> > I have not yet checked to make sure that this works properly without all of the additional options i include when creating an unattend file, but i think `<HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>` should be what you need
> > I would also suggest adding `<HideOnlineAccountScreens>true</HideOnlineAccountScreens>` as that will bypass it even if you accidently leave a network cable connected (if the computer is connected using a wired connection)
> > it should look something like this:
> > ```
> >     <settings pass="oobeSystem">
> >         <component name="Microsoft-Windows-Shell-Setup" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="nonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
> >             <OOBE>
> >                 <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
> >                 <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
> >             </OOBE>
> >         </component>
> >     </settings>
> > ```
> 
> Do you add this file in the root of the ISO folder or in the Sources folder?

For Autounattend files, you would add it to the root of the ISO folder (the installer automatically looks for Autounattend files on all removable drives)
the file must be called `Autounattend.xml` for the installer to find it
If you have specified a product key, i would suggest treating it as an Autounattend file

**Comment by engineer22** [comment]
@pineapple63 so I should probably use AutoUnattend.xml and put it in the root folder of the ISO.

I have added a folder in the $OEM$ folder that it will activate Windows 11 and it had a generic key. I should also add it kn tje AutoUnattend.xml file where the Key is specified?

**Comment by pineapple63** [comment]
@engineer22 
With the file that @pbatard linked to, that would need to be placed in the root of the ISO, and you will need to add the generic key to the file
If you want to use your own unattend.xml file that is placed in the OEM folder, using what i provided above to bypass the microsoft account, then it should work (if you do this, do not rename it to autounattend), just make sure to put what i provided in the correct place

If you need further help i would suggest going to windows forums for help, rather than asking for help here

**Comment by engineer22** [comment]
Hey @pineapple63 I didn't see your ladt reply until now.

I figured it out and everything works correctly. I was able to download the AutoUnattend.xml file from the MCT project and mount the boot.wim Index 2 so I can place the file inside it.

Added my OEM folder in the sources folder too and now I have a clean ISO that bypasses all restrictions, the Microsoft Account requirement and permanently activates Windows.
